### PR TITLE
cli: add app:build app:serve and use in app

### LINKED ISSUE #286: add app:build, app:serve to cli
There should be `app:build/serve/test/lint` commands that are used by `packages/app` and any other instance of backstage.

App package.json:

```
"scripts": {
  "build": "backstage-cli app:build",
  ...
}
```

Not sure if we want to base this on `react-scripts` or a custom webpack. Leaning towards custom webpack, especially since we have the custom optimization.

### PR DESCRIPTION
**Pull request by Rugvip**
Fixes #286

### CODE REVIEW

**Comment by Rugvip** [comment]
Unlocks work on #287 since we can iterate on the build without updating the template

File: packages/cli/src/commands/watch-deps/index.ts [hunk 1]
```diff
@@ -31,16 +31,9 @@ const PACKAGE_BLACKLIST = [
 
 const WATCH_LOCATIONS = ['package.json', 'src', 'assets'];
```
Comment (freben) [inline packages/cli/src/commands/watch-deps/index.ts]: add `dev` here too, later, after we add `plugin:serve`, right?

**Review by freben — APPROVED** [review]

File: packages/cli/src/commands/watch-deps/index.ts (hunk 1 above)
Comment (Rugvip) [reply packages/cli/src/commands/watch-deps/index.ts]: dev would only be for serving the plugin itself, shouldn't affect build output of the plugin

Comment (freben) [reply packages/cli/src/commands/watch-deps/index.ts]: i thought this function would be used inside `backstage-cli plugin:serve`, whose rendered output does depend on changes in `dev` (if that's even how we solve that one). but i may be conflating things

Comment (Rugvip) [reply packages/cli/src/commands/watch-deps/index.ts]: This is for watching other packages though, plugin:serve will have it's own local watch mode based on webpack. The locations are the ones we want to watch inside other packages, such as core, or plugins when we're serving the app.

Comment (freben) [reply packages/cli/src/commands/watch-deps/index.ts]: I see, thanks for clarifying!

**Pull request merged**